nntrainer/layers/flatten_layer.cpp

```diff
@@ -43,6 +46,8 @@ void FlattenLayer::setProperty(const std::vector<std::string> &values) {
 }
 
 void FlattenLayer::exportTo(Exporter &exporter,
-                            const ExportMethods &method) const {}
+                            const ExportMethods &method) const {
+  exporter.saveResult(reshape_props, method, this);
+}
```

Comment:
This change makes some test failed. 
```bash
[----------] Global test environment tear-down
[==========] 303 tests from 2 test cases ran. (8745 ms total)
[  PASSED  ] 263 tests.
[  FAILED  ] 40 tests, listed below:
[  FAILED  ] nntrainerModelAutoTests/nntrainerModelTest.model_test_save_load_compare/conv_1x1, where GetParam() = (32-byte object <80-6F E4-31 55-56 00-00 AF-AF AF-AF AF-AF AF-AF BF-8E 05-31 55-56 00-00 68-8E 05-31 55-56 00-00>, "conv_1x1", Shape: 3:1:1:10
, 10, 7)
[  FAILED  ] nntrainerModelAutoTests/nntrainerModelTest.model_test_save_load_compare/conv_input_matches_kernel, where GetParam() = (32-byte object <C0-AC E4-31 55-56 00-00 AF-AF AF-AF AF-AF AF-AF BF-8E 05-31 55-56 00-00 68-8E 05-31 55-56 00-00>, "conv_input_matches_kernel", Shape: 3:1:1:10
, 10, 7)
[  FAILED  ] nntrainerModelAutoTests/nntrainerModelTest.model_test_save_load_compare/conv_basic, where GetParam() = (32-byte object <70-91 E4-31 55-56 00-00 AF-AF AF-AF AF-AF AF-AF BF-8E 05-31 55-56 00-00 68-8E 05-31 55-56 00-00>, "conv_basic", Shape: 3:1:1:10
, 10, 7)
[  FAILED  ] nntrainerModelAutoTests/nntrainerModelTest.model_test_save_load_compare/conv_same_padding, where GetParam() = (32-byte object <A0-F9 E5-31 55-56 00-00 AF-AF AF-AF AF-AF AF-AF BF-8E 05-31 55-56 00-00 68-8E 05-31 55-56 00-00>, "conv_same_padding", Shape: 3:1:1:10
, 10, 7)
[  FAILED  ] nntrainerModelAutoTests/nntrainerModelTest.model_test_save_load_compare/conv_multi_stride, where GetParam() = (32-byte object <A0-F9 E5-31 55-56 00-00 AF-AF AF-AF AF-AF AF-AF BF-8E 05-31 55-56 00-00 68-8E 05-31 55-56 00-00>, "conv_multi_stride", Shape: 3:1:1:10
, 10, 7)
[  FAILED  ] nntrainerModelAutoTests/nntrainerModelTest.model_test_save_load_compare/conv_uneven_strides, where GetParam() = (32-byte object <A0-F9 E5-31 55-56 00-00 AF-AF AF-AF AF-AF AF-AF BF-8E 05-31 55-56 00-00 68-8E 05-31 55-56 00-00>, "conv_uneven_strides", Shape: 3:1:1:10
, 10, 7)
[  FAILED  ] nntrainerModelAutoTests/nntrainerModelTest.model_test_save_load_compare/conv_uneven_strides2, where GetParam() = (32-byte object <A0-F9 E5-31 55-56 00-00 AF-AF AF-AF AF-AF AF-AF BF-8E 05-31 55-56 00-00 68-8E 05-31 55-56 00-00>, "conv_uneven_strides2", Shape: 3:1:1:10
, 10, 7)
[  FAILED  ] nntrainerModelAutoTests/nntrainerModelTest.model_test_save_load_compare/conv_uneven_strides3, where GetParam() = (32-byte object <A0-F9 E5-31 55-56 00-00 AF-AF AF-AF AF-AF AF-AF BF-8E 05-31 55-56 00-00 68-8E 05-31 55-56 00-00>, "conv_uneven_strides3", Shape: 3:1:1:10
, 10, 7)
[  FAILED  ] nntrainerModelAutoTests/nntrainerModelTest.model_test_save_load_compare/conv_bn, where GetParam() = (32-byte object <A0-F9 E5-31 55-56 00-00 AF-AF AF-AF AF-AF AF-AF BF-8E 05-31 55-56 00-00 68-8E 05-31 55-56 00-00>, "conv_bn", Shape: 3:1:1:10
, 10, 7)
[  FAILED  ] nntrainerModelAutoTests/nntrainerModelTest.model_test_save_load_compare/conv_same_padding_multi_stride, where GetParam() = (32-byte object <A0-F9 E5-31 55-56 00-00 AF-AF AF-AF AF-AF AF-AF BF-8E 05-31 55-56 00-00 68-8E 05-31 55-56 00-00>, "conv_same_padding_multi_stride", Shape: 3:1:1:10
, 10, 7)
[  FAILED  ] nntrainerModelAutoTests/nntrainerModelTest.model_test_save_load_compare/pooling_max_same_padding, where GetParam() = (32-byte object <A0-F9 E5-31 55-56 00-00 AF-AF AF-AF AF-AF AF-AF BF-8E 05-31 55-56 00-00 68-8E 05-31 55-56 00-00>, "pooling_max_same_padding", Shape: 3:1:1:10
, 10, 7)
[  FAILED  ] nntrainerModelAutoTests/nntrainerModelTest.model_test_save_load_compare/pooling_max_same_padding_multi_stride, where GetParam() = (32-byte object <A0-F9 E5-31 55-56 00-00 AF-AF AF-AF AF-AF AF-AF BF-8E 05-31 55-56 00-00 68-8E 05-31 55-56 00-00>, "pooling_max_same_padding_multi_stride", Shape: 3:1:1:10
, 10, 7)
[  FAILED  ] nntrainerModelAutoTests/nntrainerModelTest.model_test_save_load_compare/pooling_max_valid_padding, where GetParam() = (32-byte object <A0-F9 E5-31 55-56 00-00 AF-AF AF-AF AF-AF AF-AF BF-8E 05-31 55-56 00-00 68-8E 05-31 55-56 00-00>, "pooling_max_valid_padding", Shape: 3:1:1:10
, 10, 7)
[  FAILED  ] nntrainerModelAutoTests/nntrainerModelTest.model_test_save_load_compare/pooling_avg_same_padding, where GetParam() = (32-byte object <A0-F9 E5-31 55-56 00-00 AF-AF AF-AF AF-AF AF-AF BF-8E 05-31 55-56 00-00 68-8E 05-31 55-56 00-00>, "pooling_avg_same_padding", Shape: 3:1:1:10
, 10, 7)
[  FAILED  ] nntrainerModelAutoTests/nntrainerModelTest.model_test_save_load_compare/pooling_avg_same_padding_multi_stride, where GetParam() = (32-byte object <A0-F9 E5-31 55-56 00-00 AF-AF AF-AF AF-AF AF-AF BF-8E 05-31 55-56 00-00 68-8E 05-31 55-56 00-00>, "pooling_avg_same_padding_multi_stride", Shape: 3:1:1:10
, 10, 7)
[  FAILED  ] nntrainerModelAutoTests/nntrainerModelTest.model_test_save_load_compare/pooling_avg_valid_padding, where GetParam() = (32-byte object <A0-F9 E5-31 55-56 00-00 AF-AF AF-AF AF-AF AF-AF BF-8E 05-31 55-56 00-00 68-8E 05-31 55-56 00-00>, "pooling_avg_valid_padding", Shape: 3:1:1:10
, 10, 7)
[  FAILED  ] nntrainerModelAutoTests/nntrainerModelTest.model_test_save_load_compare/pooling_global_avg, where GetParam() = (32-byte object <A0-F9 E5-31 55-56 00-00 AF-AF AF-AF AF-AF AF-AF BF-8E 05-31 55-56 00-00 68-8E 05-31 55-56 00-00>, "pooling_global_avg", Shape: 3:1:1:10
, 10, 7)
[  FAILED  ] nntrainerModelAutoTests/nntrainerModelTest.model_test_save_load_compare/pooling_global_max, where GetParam() = (32-byte object <A0-F9 E5-31 55-56 00-00 AF-AF AF-AF AF-AF AF-AF BF-8E 05-31 55-56 00-00 68-8E 05-31 55-56 00-00>, "pooling_global_max", Shape: 3:1:1:10
, 10, 7)
[  FAILED  ] nntrainerModelAutoTests/nntrainerModelTest.model_test_save_load_compare/mnist_conv_cross, where GetParam() = (32-byte object <A0-F9 E5-31 55-56 00-00 AF-AF AF-AF AF-AF AF-AF BF-8E 05-31 55-56 00-00 68-8E 05-31 55-56 00-00>, "mnist_conv_cross", Shape: 3:1:1:10
, 10, 7)
[  FAILED  ] nntrainerModelAutoTests/nntrainerModelTest.model_test_save_load_compare/mnist_conv_cross_one_input, where GetParam() = (32-byte object <A0-F9 E5-31 55-56 00-00 AF-AF AF-AF AF-AF AF-AF BF-8E 05-31 55-56 00-00 68-8E 05-31 55-56 00-00>, "mnist_conv_cross_one_input", Shape: 1:1:1:10
, 10, 7)
[  FAILED  ] nntrainerModelAutoTests/nntrainerModelTest.model_test_save_load_verify/conv_1x1, where GetParam() = (32-byte object <E0-1E E8-31 55-56 00-00 AF-AF AF-AF AF-AF AF-AF BF-8E 05-31 55-56 00-00 68-8E 05-31 55-56 00-00>, "conv_1x1", Shape: 3:1:1:10
, 10, 7)
[  FAILED  ] nntrainerModelAutoTests/nntrainerModelTest.model_test_save_load_verify/conv_input_matches_kernel, where GetParam() = (32-byte object <00-3B E8-31 55-56 00-00 AF-AF AF-AF AF-AF AF-AF BF-8E 05-31 55-56 00-00 68-8E 05-31 55-56 00-00>, "conv_input_matches_kernel", Shape: 3:1:1:10
, 10, 7)
[  FAILED  ] nntrainerModelAutoTests/nntrainerModelTest.model_test_save_load_verify/conv_basic, where GetParam() = (32-byte object <00-28 E8-31 55-56 00-00 AF-AF AF-AF AF-AF AF-AF BF-8E 05-31 55-56 00-00 68-8E 05-31 55-56 00-00>, "conv_basic", Shape: 3:1:1:10
, 10, 7)
[  FAILED  ] nntrainerModelAutoTests/nntrainerModelTest.model_test_save_load_verify/conv_same_padding, where GetParam() = (32-byte object <80-8A E9-31 55-56 00-00 AF-AF AF-AF AF-AF AF-AF BF-8E 05-31 55-56 00-00 68-8E 05-31 55-56 00-00>, "conv_same_padding", Shape: 3:1:1:10
, 10, 7)
[  FAILED  ] nntrainerModelAutoTests/nntrainerModelTest.model_test_save_load_verify/conv_multi_stride, where GetParam() = (32-byte object <80-8A E9-31 55-56 00-00 AF-AF AF-AF AF-AF AF-AF BF-8E 05-31 55-56 00-00 68-8E 05-31 55-56 00-00>, "conv_multi_stride", Shape: 3:1:1:10
, 10, 7)
[  FAILED  ] nntrainerModelAutoTests/nntrainerModelTest.model_test_save_load_verify/conv_uneven_strides, where GetParam() = (32-byte object <80-8A E9-31 55-56 00-00 AF-AF AF-AF AF-AF AF-AF BF-8E 05-31 55-56 00-00 68-8E 05-31 55-56 00-00>, "conv_uneven_strides", Shape: 3:1:1:10
, 10, 7)
[  FAILED  ] nntrainerModelAutoTests/nntrainerModelTest.model_test_save_load_verify/conv_uneven_strides2, where GetParam() = (32-byte object <80-8A E9-31 55-56 00-00 AF-AF AF-AF AF-AF AF-AF BF-8E 05-31 55-56 00-00 68-8E 05-31 55-56 00-00>, "conv_uneven_strides2", Shape: 3:1:1:10
, 10, 7)
[  FAILED  ] nntrainerModelAutoTests/nntrainerModelTest.model_test_save_load_verify/conv_uneven_strides3, where GetParam() = (32-byte object <80-8A E9-31 55-56 00-00 AF-AF AF-AF AF-AF AF-AF BF-8E 05-31 55-56 00-00 68-8E 05-31 55-56 00-00>, "conv_uneven_strides3", Shape: 3:1:1:10
, 10, 7)
[  FAILED  ] nntrainerModelAutoTests/nntrainerModelTest.model_test_save_load_verify/conv_bn, where GetParam() = (32-byte object <80-8A E9-31 55-56 00-00 AF-AF AF-AF AF-AF AF-AF BF-8E 05-31 55-56 00-00 68-8E 05-31 55-56 00-00>, "conv_bn", Shape: 3:1:1:10
, 10, 7)
[  FAILED  ] nntrainerModelAutoTests/nntrainerModelTest.model_test_save_load_verify/conv_same_padding_multi_stride, where GetParam() = (32-byte object <80-8A E9-31 55-56 00-00 AF-AF AF-AF AF-AF AF-AF BF-8E 05-31 55-56 00-00 68-8E 05-31 55-56 00-00>, "conv_same_padding_multi_stride", Shape: 3:1:1:10
, 10, 7)
[  FAILED  ] nntrainerModelAutoTests/nntrainerModelTest.model_test_save_load_verify/pooling_max_same_padding, where GetParam() = (32-byte object <80-8A E9-31 55-56 00-00 AF-AF AF-AF AF-AF AF-AF BF-8E 05-31 55-56 00-00 68-8E 05-31 55-56 00-00>, "pooling_max_same_padding", Shape: 3:1:1:10
, 10, 7)
[  FAILED  ] nntrainerModelAutoTests/nntrainerModelTest.model_test_save_load_verify/pooling_max_same_padding_multi_stride, where GetParam() = (32-byte object <80-8A E9-31 55-56 00-00 AF-AF AF-AF AF-AF AF-AF BF-8E 05-31 55-56 00-00 68-8E 05-31 55-56 00-00>, "pooling_max_same_padding_multi_stride", Shape: 3:1:1:10
, 10, 7)
[  FAILED  ] nntrainerModelAutoTests/nntrainerModelTest.model_test_save_load_verify/pooling_max_valid_padding, where GetParam() = (32-byte object <80-8A E9-31 55-56 00-00 AF-AF AF-AF AF-AF AF-AF BF-8E 05-31 55-56 00-00 68-8E 05-31 55-56 00-00>, "pooling_max_valid_padding", Shape: 3:1:1:10
, 10, 7)
[  FAILED  ] nntrainerModelAutoTests/nntrainerModelTest.model_test_save_load_verify/pooling_avg_same_padding, where GetParam() = (32-byte object <80-8A E9-31 55-56 00-00 AF-AF AF-AF AF-AF AF-AF BF-8E 05-31 55-56 00-00 68-8E 05-31 55-56 00-00>, "pooling_avg_same_padding", Shape: 3:1:1:10
, 10, 7)
[  FAILED  ] nntrainerModelAutoTests/nntrainerModelTest.model_test_save_load_verify/pooling_avg_same_padding_multi_stride, where GetParam() = (32-byte object <80-8A E9-31 55-56 00-00 AF-AF AF-AF AF-AF AF-AF BF-8E 05-31 55-56 00-00 68-8E 05-31 55-56 00-00>, "pooling_avg_same_padding_multi_stride", Shape: 3:1:1:10
, 10, 7)
[  FAILED  ] nntrainerModelAutoTests/nntrainerModelTest.model_test_save_load_verify/pooling_avg_valid_padding, where GetParam() = (32-byte object <80-8A E9-31 55-56 00-00 AF-AF AF-AF AF-AF AF-AF BF-8E 05-31 55-56 00-00 68-8E 05-31 55-56 00-00>, "pooling_avg_valid_padding", Shape: 3:1:1:10
, 10, 7)
[  FAILED  ] nntrainerModelAutoTests/nntrainerModelTest.model_test_save_load_verify/pooling_global_avg, where GetParam() = (32-byte object <80-8A E9-31 55-56 00-00 AF-AF AF-AF AF-AF AF-AF BF-8E 05-31 55-56 00-00 68-8E 05-31 55-56 00-00>, "pooling_global_avg", Shape: 3:1:1:10
, 10, 7)
[  FAILED  ] nntrainerModelAutoTests/nntrainerModelTest.model_test_save_load_verify/pooling_global_max, where GetParam() = (32-byte object <80-8A E9-31 55-56 00-00 AF-AF AF-AF AF-AF AF-AF BF-8E 05-31 55-56 00-00 68-8E 05-31 55-56 00-00>, "pooling_global_max", Shape: 3:1:1:10
, 10, 7)
[  FAILED  ] nntrainerModelAutoTests/nntrainerModelTest.model_test_save_load_verify/mnist_conv_cross, where GetParam() = (32-byte object <80-8A E9-31 55-56 00-00 AF-AF AF-AF AF-AF AF-AF BF-8E 05-31 55-56 00-00 68-8E 05-31 55-56 00-00>, "mnist_conv_cross", Shape: 3:1:1:10
, 10, 7)
[  FAILED  ] nntrainerModelAutoTests/nntrainerModelTest.model_test_save_load_verify/mnist_conv_cross_one_input, where GetParam() = (32-byte object <80-8A E9-31 55-56 00-00 AF-AF AF-AF AF-AF AF-AF BF-8E 05-31 55-56 00-00 68-8E 05-31 55-56 00-00>, "mnist_conv_cross_one_input", Shape: 1:1:1:10
, 10, 7)

40 FAILED TESTS
-------
```

@jijoongmoon Could you let me know about this? I need this change because `void Exporter::saveTflResult(const std::tuple<props::TargetShape> &props,
                             const FlattenLayer *self)` function should be called.

Reply:
This will be fixed after you change the finalize.